versioned_docs/version-8.5/self-managed/operational-guides/troubleshooting/troubleshooting.md

```diff
@@ -66,3 +66,81 @@ A gateway timeout can occur if the headers of a response are too big (for exampl
+    
+    ```bash
+    ./checks/kube/deployment.sh -n camunda-primary -d camunda -c "zeebe,zeebe-gateway,web-modeler"
+    ```
+
+    #### Connectivity Check (`./checks/kube/connectivity.sh`)
+    
+    This script verifies Kubernetes connectivity and associated configuration, checking for the presence of services and ingresses that conform to the required specifications.
+    
+    ```bash
+    ./checks/kube/connectivity.sh -n camunda-primary
+    ```
+
+- **Zeebe Connectivity Scripts**: These scripts focus on verifying the connectivity and health of Zeebe components within your deployment. You can check token generation, gRPC connectivity, and other essential aspects of your Zeebe setup.
+
+    #### gRPC Zeebe Check (`./checks/zeebe/connectivity.sh`)
+    
+    This script verifies connectivity to a Zeebe instance using HTTP/2 and gRPC protocols, providing insights into the health and status of your Zeebe deployment.
+    
+    ```bash
+    ./checks/zeebe/connectivity.sh -a https://local.distro.example.com/auth/realms/camunda-platform/protocol/openid-connect/token -i myclientid -s 0Rn28VrQxGNxowrCWe6wbujwFghO4990 -u zeebe.distro.example.com -H zeebe.local.distro.example.com:443
+    ```
+
+  You can find here [how to register your application on Identity](https://github.com/camunda-community-hub/camunda-8-examples/blob/main/payment-example-process-application/kube/README.md#4-generating-an-m2m-token-for-our-application).
+
+### Interpretation of the results
+
+Each script produces output indicating the status of individual checks, which can be either '[OK]' or '[KO]'.
+While the scripts continue execution even if a check fails, it may be necessary to review the logs to identify the failed element.
+
+At the end of each script, a global check status is provided, indicating whether any tests failed and the corresponding error code. For example:
+
+```
+[KO] ./checks/zeebe/connectivity.sh: At least one of the tests failed (error code: 5).
+```
+
+### Handling Errors
+
+If a check fails, it indicates a deviation from the expected configuration on a normal setup. Resolving the error involves studying the failed check and applying the best practices outlined in the documentation (use the search feature to find the associated recommendation for a failed check).
+
+For example:
+
+```
+[KO] None of the ingresses contain the annotation nginx.ingress.kubernetes.io/backend-protocol: GRPC, which is required for Zeebe ingress.
+```
+
+The error message suggests adjusting the ingress configuration to include the required annotation. One can also explore the source of the script to have a better understanding of the reason for the failure.
+
+Please also note that sometimes, some checks may not be applicable to your setup if it's custom (e.g., the ingress may be different).
```

Comment:
`e.g.` replace with `for example`, see [related message](https://camunda.slack.com/archives/C026U8GBNSW/p1709922869101609)